### ISSUE TITLE
difficult debugging of static files sources

### ISSUE THREAD

**Issue by lucamorelli** [aspnet/Home#347]
I'm using ctp 6 and release feed.
I created a web starter application and added the **StaticFiles** sources packages cloning tre repository and adding the source directory to the global.json.
All works fine, but I have a problem debugging the sources:when I add breakpoints to the configuration methods like **StaticFilesMiddleWare** all is fine, but if I place a breakpoint inside the Invoke method or inside the code is called processing a request, when I reach the breakpoint and press F10, seems that it's reached multiple times before I can go to the next step.
I'm quite sure the problem (I checked in the context the requested file) is that multiple request are done  at the same time, and visual studio stops at every step of every request. This is good but, as you can imagine makes very difficult to debug it, because I can't debug a single request.
I tried this using both IS express and Web command with the same result.
Exists a way to reduce the number of requests processed at the same time, or configure Vs to follow just one single request.

To repro:

1. create a web starter application.
2. clone the staticfiles repositories sources (OUTSIDE the solution folder)
3. add the static files source directory to the global.json and rebuild.
4. add a breakpoint inside the **Invoke** method of the **StaticFileMiddleware** class and run the app.
5. reach the breakpoint and use f10 to debug the method call

**Comment by Eilon** [comment]
In VS you can create a conditional breakpoint that gets hit only under whatever conditions you specify (you can type in an expression that evaluates, and when it's true, the breakpoint hits). You can read more about them here: http://blogs.msdn.com/b/visualstudioalm/archive/2013/10/11/conditional-breakpoints.aspx

You could set a condition that perhaps looks at the current request's URL and compares it to the particular URL you are looking at. So the condition would be something like:

:ballot_box_with_check: Conditional Expression | Is True | [context.Request.Url == "/someUrlToCompare"]

(The syntax is totally wrong, but the general point should work.)

**Comment by davidfowl** [comment]
It's probably hit multiple times because there's multiple requests

**Comment by Eilon** [comment]
Yeah no doubt the browser is asking for favicon.ico, and various other things.

**Comment by lucamorelli** [comment]
well, that's strange ... i tried setting a breakpoint using the content.**Request.path.Value** and started to work fine using **IIS express**. After a few tests changed behavior, stopping to break at the condition. I removed the condition, and I see that the breakpoint in the **Invoke** method just at first time, with the Request.Path = "/", and after continues to run, but the web page is shown correctly, so I think is executed multiple times. Closed vs and reopened, nothing happens. The path is different from the one I initially set, so isn't the case that the brealpoint is filtered but is not shown in vs. I tried to change from IIS express to **web command** and started to work again. After a few tests I switched back to IIS express and started to work again.
Now seems to work. I have the doubt is not a VS problem, but the browser cache the files and sometime doesn't request files.
